### PR TITLE
added the wufoo shortcode from .com

### LINKED ISSUE #1180: Wufoo Shortcode
There's a Wufoo shortcode on WordPress.com that should be included in Jetpack.

### PR DESCRIPTION
**Pull request by jessefriedman**
Porting over Wufoo shortcodes from WPCOM verbatim.

This file has been added to the build script.  

fixes #1180

Example shortcode for testing: 
`[wufoo username="jesserfriedman" formhash="zk8za6p0ma7g2j" autoresize="true" height="260" header="show" ssl="true”]`

### CODE REVIEW

File: modules/shortcodes/wufoo.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	), $atts ) );
+
+	//Check username and formhash to ensure they only have alphanumeric characters or underscores, and aren't empty
+	if ( !preg_match( "/^[a-zA-Z0-9_]+$/", $username) || !preg_match( "/^[a-zA-Z0-9_]+$/", $formhash ) ) {
+
+		/**
+		 * Return an error to the users with instructions if one of these params is invalid
+		 * They don't have default values because they are user/form-specific
+		 */
+		return "
```
Comment (jeherve) [inline modules/shortcodes/wufoo.php]: This would need to be translated, I believe.

File: modules/shortcodes/wufoo.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	$iframe_embed .= 'rel="nofollow">Fill out my Wufoo form!</a></iframe>';
+
+	do_action( 'jetpack_stats_extra', 'embeds', 'wufoo' );
+	/**
+	 * Return embed in JS and iframe
+	 */
+	return "$js_embed <noscript> $iframe_embed </noscript>";
+}
+
+add_shortcode('wufoo', 'wufoo_shortcode');
```
Comment (jeherve) [inline modules/shortcodes/wufoo.php]: It seems like we miss some spaces here, as per the [core coding standards](https://make.wordpress.org/core/handbook/coding-standards/php/#space-usage).

File: modules/shortcodes/wufoo.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	if ( !preg_match( "/^[a-zA-Z0-9_]+$/", $username) || !preg_match( "/^[a-zA-Z0-9_]+$/", $formhash ) ) {
+
+		/**
+		 * Return an error to the users with instructions if one of these params is invalid
+		 * They don't have default values because they are user/form-specific
+		 */
+		return "
+			<div style='border: 20px solid red; border-radius: 40px; padding: 40px; margin: 50px 0 70px;'>
+				<h3>Uh oh!</h3>
+				<p style='margin: 0;'>Something is wrong with your Wufoo shortcode. If you copy and paste it from the <a href='http://wufoo.com/docs/code-manager/'>Wufoo Code Manager</a>, you should be golden.</p>
```
Comment (dereksmart) [inline modules/shortcodes/wufoo.php]: Accidently deleted @jeherve's comment, but he's right in that this will need to be translated

**Comment by jeherve** [comment]
@jessefriedman Any chance you could make the changes suggested above? It should be good to merge after that.

**Comment by zinigor** [comment]
Let's hold on for a while, I want to do the right thing and make those changes synchronously. I will force-update this PR once I have prepared the code to be synced from wpcom.

**Comment by jessefriedman** [comment]
@jeherve will do

**Comment by dereksmart** [comment]
I've updated the dotcom code so it will merge over easier.  I've rebased this PR so that it reflects the dotcom code exactly.  I've tested and everything seems good to go.  Just waiting for one more tester.

**Comment by jessefriedman** [comment]
I've tested and it works for me

**Pull request merged**